pkg/download/params_middleware.go

```diff
@@ -0,0 +1,20 @@
+package download
+
+import (
+	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/paths"
+)
+
+// ParamsMiddleware fetches module and version params from c and calls next with "value"
+// and "version" strings in the context. Access them with c.Value("module") and c.Value("version")
+func ParamsMiddleware(next buffalo.Handler) buffalo.Handler {
```

Comment:
Why do we need middleware for this if we can just call path.GetAllParams from inside a handler?

Reply:
PS. if it's the error handling part, paths.GetAllParams shouldn't really return an error. It can safely panic because that means we got something seriously screwed up in our http paths.